pkg/vsphere/disk/disk_manager.go

```diff
@@ -421,3 +428,48 @@ func (m *Manager) devicePathByURI(op trace.Operation, datastoreURI *object.Datas
 
 	return sysPath, nil
 }
+
+func (m *Manager) InUse(op trace.Operation, config *VirtualDiskConfig) (*vm.VirtualMachine, error) {
```

Comment:
It is not possible to attach the same vmdk to multiple vms with our current usage (means there is a way but we are not using shared disks). 

I'm not sure what you mean by the second part but I'm guessing you are suggesting checking connected field when we find a match before returning it. AFAIK it doesn't matter whether the disk is connected or not as we still cannot attach it to some other vm without detaching first.